fml/task_source.cc

```diff
@@ -0,0 +1,103 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#define FML_USED_ON_EMBEDDER
+
+#include "flutter/fml/task_source.h"
+
+namespace fml {
+
+TaskSource::TaskSource(TaskQueueId task_queue_id)
+    : task_queue_id_(task_queue_id) {}
+
+TaskSource::~TaskSource() {
+  ShutDown();
+}
+
+void TaskSource::ShutDown() {
+  primary_task_queue_ = {};
+  secondary_task_queue_ = {};
+}
+
+void TaskSource::RegisterTask(const DelayedTask& task) {
+  switch (task.GetTaskSourceGrade()) {
+    case TaskSourceGrade::kUserInteraction:
+      primary_task_queue_.push(task);
+      break;
```

Comment:
Nit: Here and later, you can use [[fallthrough]] if the compiler troubles you about the missing break. You call though.